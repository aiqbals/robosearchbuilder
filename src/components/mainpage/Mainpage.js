import React, { Component } from 'react';

import Clock from '../clock/Clock';
import CardList from '../cardlist/CardList';
import Scroll from '../scroll/Scroll';
import Searchbox from '../searchbox/Searchbox';
//import {robots} from './robots';
import ErrorBoundry from '../errorboundry/ErrorBoundry';
import './Mainpage.css';


class Mainpage extends Component {

    componentDidMount(){
        this.props.onRequestRobots();
    }

    filteredRobot = () => {
        return this.props.robots.filter( robot => {
        return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        })
    }  

    render() {
        const { searchField, onSearchChange, robots, isPending } = this.props;
        /* const filteredRobot = robots.filter( robot => {
            return robot.name.toLowerCase().includes(searchField.toLowerCase());
        }); */

        return (
            <div className='tc'>
                    <h1 className='f2'> RoboBuild Search! </h1>
                    <Searchbox searchChange={onSearchChange} />
                    <Clock />
                    <Scroll>
                        { isPending ? <h1> Loading...</h1> :
                            <ErrorBoundry>
                                <CardList robots={this.filteredRobot()} />
                            </ErrorBoundry>
                        }
                    </Scroll>
                </div> 
        );  
    }
}

export default Mainpage;