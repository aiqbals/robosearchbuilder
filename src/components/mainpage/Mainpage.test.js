import { shallow } from 'enzyme';
import React from 'react';
import Mainpage from './Mainpage';

let wrapper;
beforeEach( () => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }
    wrapper = shallow(<Mainpage {...mockProps} />)
})

it('Renders Mainpage without crashing', () => {
    expect(wrapper).toMatchSnapshot()
})

it('Filters robots correctly', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'Asf'
        }],
        searchField: 'a',
        isPending: false
    }
    const wrapper2 = shallow(<Mainpage {...mockProps2} />)
    expect(wrapper2.instance().filteredRobot()).toEqual([
        {
            id: 3,
            name: 'Asf'
        }
    ])
    //instance() property comes with that gives us access to all methods in Mainpage
})