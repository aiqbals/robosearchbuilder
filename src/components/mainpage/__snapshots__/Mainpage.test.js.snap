// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Mainpage without crashing 1`] = `ShallowWrapper {}`;
